Xamarin.PropertyEditing.Mac/Controls/RequestResource/ResourceTableDelegate.cs

```diff
@@ -0,0 +1,171 @@
+		const string valueIdentifier = "value";
+
+		public event EventHandler ResourceSelected;
+
+		private nint previousRow = -1;
+
+		public ResourceTableDelegate (ResourceTableDataSource datasource)
+		{
+			this.datasource = datasource;
+		}
+
+		// the table is looking for this method, picks it up automagically
+		public override NSView GetViewForItem (NSTableView tableView, NSTableColumn tableColumn, nint row)
+		{
+			var resource = datasource.ViewModel.Resources [(int)row] as Resource;
+
+			// Setup view based on the column
+			switch (tableColumn.Identifier) {
+				case RequestResourcePanel.ResourceImageColId:
+					if (resource.Source.Type != ResourceSourceType.Application) {
+						var iconView = (NSImageView)tableView.MakeView (iconIdentifier, this);
+						if (iconView == null) {
+							iconView = new NSImageView {
+								Image = PropertyEditorPanel.ThemeManager.GetImageForTheme ("resource-editor-32"),
+								ImageScaling = NSImageScale.None,
+								Identifier = iconIdentifier,
+							};
+						}
+						return iconView;
+					}
+
+					// If we get here its a local resource
+					return new NSView ();
+
+				case RequestResourcePanel.ResourceTypeColId:
+					var typeView = (UnfocusableTextField)tableView.MakeView (typeIdentifier, this);
+					if (typeView == null) {
+						typeView = new UnfocusableTextField {
+							Identifier = typeIdentifier,
+							BackgroundColor = NSColor.Clear,
+						};
+					}
+
+					typeView.StringValue = resource.Source.Name;
+					return typeView;
+
+				case RequestResourcePanel.ResourceNameColId:
+					var nameView = (UnfocusableTextField)tableView.MakeView (nameIdentifier, this);
+					if (nameView == null) {
+						nameView = new UnfocusableTextField {
+							BackgroundColor = NSColor.Clear,
+							Identifier = nameIdentifier,
+						};
+					}
+
+					nameView.StringValue = resource.Name;
+					return nameView;
+
+				case RequestResourcePanel.ResourceValueColId:
+					var valueView = MakeValueView (resource, tableView);
+
+					// If still null we have no editor yet.
+					valueView = valueView ?? new NSView ();
+
+					return valueView;
+
+				default:
+					return base.GetViewForItem (tableView, tableColumn, row);
+			}
+		}
+
+		public override nfloat GetRowHeight (NSTableView tableView, nint row)
+		{
+			return PropertyEditorControl.DefaultControlHeight;
+		}
+
+		public override void SelectionDidChange (NSNotification notification)
+		{
+			if (notification.Object is NSTableView tableView) {
+				NSImageView iconColumn;
+				if (previousRow != -1) {
+					iconColumn = tableView.GetView (0, previousRow, false) as NSImageView;
+					iconColumn.Image = PropertyEditorPanel.ThemeManager.GetImageForTheme ("resource-editor-32");
+				}
+
+				if (tableView.SelectedRow != -1) {
+					iconColumn = tableView.GetView (0, tableView.SelectedRow, false) as NSImageView;
+					iconColumn.Image = PropertyEditorPanel.ThemeManager.GetImageForTheme ("resource-editor-32", "~sel");
```

Comment:
This throws NRE if you're opening the resource window for a property which already has a resource selected.

Reply:
Fixed.